### ISSUE TITLE
15 Last year is ignored in a wild card search. (Catherine Jury)

### ISSUE THREAD

**Issue by PatReynolds** [FreeUKGen/FreeUKRegProductIssues#1190]
Wild card searches should respect the year limits. Need to confirm issue.

(Originally part of #1189)

**Comment by PatReynolds** [comment]
Search Criteria: First Name is blank, Last Name is ALCO*. Exact matching is enabled.
Search Criteria: Start Year is 1820. End Year is 1840.
All three record types — Baptisms, Burials and Marriages — were included.
Search Criteria: Search additional places near Eryholme in Yorkshire, North Riding

Ran 1780=1820, and last record returned was 1820.
Ran 1820-1840, and last record returned was 1840, i.e. seems to be working fine on this example.

**Comment by SteveBiggs** [comment]
I've asked Cathy to give a specific example that shows the problem.

**Comment by SteveBiggs** [comment]
Here is Cathy's example:

The specific search as regards the Last Year being ignored:

Surname ‘bu*’, Last Year 1710 (first year is blank- I think this is what is causing the ‘Last Year’ to be ignored), County ‘Cornwall’, Place ‘Stithians’.

E0, U4, S5, C6:P15

**Comment by richpomfret** [comment]
Needs a bit of further testing.

**Comment by PatReynolds** [comment]
Al*, baptisms, North Yorks, Northallerton, end year 1710 works as expected
All types of record, otherwise same search, also works.

**Comment by SteveBiggs** [comment]
Cathy's search result:
![image](https://user-images.githubusercontent.com/25303607/54546960-b6ac9400-499c-11e9-99e0-6b3a59cf3e7e.png)
gives results up to 1901.

**Comment by SteveBiggs** [comment]
Pat's test also fails:
![image](https://user-images.githubusercontent.com/25303607/54547266-4e11e700-499d-11e9-9127-03ee18cc09cf.png)
![image](https://user-images.githubusercontent.com/25303607/54547337-697cf200-499d-11e9-87ee-450c676be5e4.png)

**Comment by Vino-S** [comment]
in the above image it says start year as 1710

**Comment by Vino-S** [comment]
Should it be End Year?

**Comment by SteveBiggs** [comment]
Yes Vino that's why I deleted my post! I put in the wrong box :-)

**Comment by SteveBiggs** [comment]
Pat's test does work correctly after all, but if you simply change the county/place to Cornwall/Stithians it fails!

**Comment by Vino-S** [comment]
Ah. Yes. Strange. Below is link to one such example
https://www.freereg.org.uk/search_queries/5c8fe4a1f493fd32153c8b68?locale=en

**Comment by PatReynolds** [comment]
Ok. Getting interesting. Altarnun (cornwall), and Ashwater and Atherington (Devon) all work as expected. But Bodmin (Cornwall) fails.

**Comment by stsccfr** [comment]
Closing as no longer reproducible.  The search engine now requires that a Start Year be entered, so if that was the cause of these problems, it no longer exists.
